metaflow/plugins/aws/awslambda/lambda_cli.py

```diff
@@ -0,0 +1,204 @@
+    LAMBDA_ROLE_ARN,
+    LAMBDA_THROTTLE_RETRIES,
+)
+from metaflow.plugins.aws.utils import sync_metadata_from_s3, TaskSpec
+
+from .lambda_runner import LambdaRunner, ensure_lambda, LambdaRuntimeException
+from metaflow.mflog import TASK_LOG_SOURCE
+
+if sys.version_info > (3, 0):
+    from typing import Dict, Optional
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.group(help="Commands related to AWS Lambda.")
+def awslambda():
+    pass
+
+
+@awslambda.command(
+    help="Execute a single task using AWS Lambda. This command "
+    "calls the top-level step command inside a AWS Lambda "
+    "job with the given options. Typically you do not "
+    "call this command directly; it is used internally "
+    "by Metaflow."
+)
+@click.argument("step-name")
+@click.argument("code-package-sha")
+@click.argument("code-package-url")
+@click.option("--memory", type=int, help="Memory requirement for AWS Lambda.")
+@click.option("--run-id", help="Passed to the top-level 'step'.")
+@click.option("--task-id", help="Passed to the top-level 'step'.")
+@click.option("--image", help="Lambda image")
+@click.option(
+    "--force_fetch_image/--no-force_fetch_image",
+    default=False,
+    help="Update lambda to run with the latest image every time.",
+)
+@click.option("--input-paths", help="Passed to the top-level 'step'.")
+@click.option("--split-index", help="Passed to the top-level 'step'.")
+@click.option("--clone-path", help="Passed to the top-level 'step'.")
+@click.option("--clone-run-id", help="Passed to the top-level 'step'.")
+@click.option(
+    "--tag", multiple=True, default=None, help="Passed to the top-level 'step'."
+)
+@click.option("--namespace", default=None, help="Passed to the top-level 'step'.")
+@click.option("--retry-count", default=0, help="Passed to the top-level 'step'.")
+@click.option(
+    "--max-user-code-retries", default=0, help="Passed to the top-level 'step'."
+)
+@click.option(
+    "--run-time-limit",
+    default=10 * 60,
+    help="Run time limit in seconds for the Lambda job. " "Default is 10 minutes.",
+)
+@click.pass_context
+def step(
+    ctx,
+    step_name,
+    code_package_sha,
+    code_package_url,
+    run_time_limit,
+    force_fetch_image,
+    memory=None,
+    image=None,
+    **kwargs
+):
+    # type: (...) -> None
+
+    def echo(msg, err):  # type (str, bool) -> None
+        ctx.obj.echo_always("%s" % (msg,), err=err)
+
+    if ctx.obj.datastore.datastore_root is None:
+        ctx.obj.datastore.datastore_root = (
+            ctx.obj.datastore.get_datastore_root_from_config(echo)
+        )
+
+    if R.use_r():
+        entrypoint = R.entrypoint()
+    else:
+        executable = ctx.obj.environment.executable(step_name)
+        entrypoint = "%s -u %s" % (executable, os.path.basename(sys.argv[0]))
+
+    top_args = " ".join(util.dict_to_cli_options(ctx.parent.parent.params))
+
+    input_paths = kwargs.get("input_paths")
+    split_vars = None
+    if input_paths:
+        max_size = 30 * 1024
```

Comment:
Do we have the same limits in AWS Lambda as we have in AWS Batch?

Reply:
no good point i should remove this